CIP-0030/README.md

```diff
@@ -247,6 +245,30 @@ Errors: `APIError`, `TxSendError`
 
 As wallets should already have this ability, we allow dApps to request that a transaction be sent through it. If the wallet accepts the transaction and tries to send it, it shall return the transaction id for the dApp to track. The wallet is free to return the `TxSendError` with code `Refused` if they do not wish to send it, or `Failure` if there was an error in sending it (e.g. preliminary checks failed on signatures).
 
+
+
+## Events
+
+In addition to the API methods that are actively called, the connector also must emit the following events. All methods events are required to be implemented.
+
+TODO: event emission method? Possible methods:
+
+1) Emitted event via `window.addEventListener(eventName , e => void)`
+2) Emitted message via `window.postMessage({eventName: string}, ...)`
+3) Some kind of callback registration i.e. `wallet.onDisconnect(() => void)` or `wallet.onEvent(eventName => void)`
+4) A combination of the two (event/message but with callback on `wallet` object as well
+5) Other?
+
+### wallet_disconnected
+
+Emitted when the user disconnects (not just changes) their current account from the page. This means that all `api` methods will return with an `APIError.Refused` error and a new `api` object must be obtained from `wallet.enable()` to continue to interact with the user's wallet.
```

Comment:
Also, when removing the dApp from the whitelist, implementation should make sure to emit the event to the page only  if the page origin (host) matches the removed host. 

App developers should only face `disconnect` events that concern their app host, and not third party hosts.